acceptance/tests/language/sensitive_data_type.rb

```diff
@@ -0,0 +1,99 @@
+       :assertions => {:expect_failure => {:refute_match => 'sekrit5', :message => 'you can always spill redacted data, if you want to'}}},
+    {:type => 'notify', :parameters => {:namevar => "6:meh", :message => '"6:${meh6}"'}, :pre_code => '$meh6=Sensitive.new("sekrit6")',
+       :assertions => [{:refute_match => 'sekrit6'}, {:assert_match => "6:#{notify_redacted}"}]},
+    {:type => 'notify', :parameters => {:namevar => "7:${Sensitive('sekrit7')}"},
+       :assertions => [{:refute_match => 'sekrit7'}, {:assert_match => "7:#{notify_redacted}"}]},
+    # unwrap(), these should be en-clair
+    {:type => 'notify', :parameters => {:namevar => "8:${unwrap(Sensitive.new('sekrit8'))}"},
+       :assertions => {:assert_match => "8:sekrit8"}},
+    {:type => 'notify', :parameters => {:namevar => "9:meh", :message => '"9:${unwrap(Sensitive.new(\'sekrit9\'))}"'},
+       :assertions => {:assert_match => "9:sekrit9"}},
+    {:type => 'notify', :parameters => {:namevar => "A:meh", :message => '"A:${unwrap($mehA)}"'}, :pre_code => '$mehA=Sensitive.new("sekritA")',
+       :assertions => {:assert_match => "A:sekritA"}},
+    {:type => 'notify', :parameters => {:namevar => "B:meh", :message => '"B:${$mehB.unwrap}"'}, :pre_code => '$mehB=Sensitive.new("sekritB")',
+       :assertions => {:assert_match => "B:sekritB"}},
+    {:type => 'notify', :parameters => {:namevar => "C:meh", :message => '"C:${$mehC.unwrap |$unwrapped| { "blk_${unwrapped}_blk" } } nonblk_${mehC}_nonblk"'}, :pre_code => '$mehC=Sensitive.new("sekritC")',
+       :assertions => {:assert_match => ["C:blk_sekritC_blk", "nonblk_#{notify_redacted}_nonblk"]}},
+    # for --show_diff
+    {:type => 'file', :parameters => {:namevar => "$pup_tmp_filename", :content => "Sensitive.new('sekritD')"}, :pre_code => "$pup_tmp_filename = if $osfamily == 'windows' { '#{tmp_filename_win}' } else { '#{tmp_filename_else}' }",
+       :assertions => [{:refute_match => 'sekritD'}, {:assert_match => /#{tmp_environment}\.txt..content. #{file_redacted}/}]},
+
+  ]
+
+  sitepp_content = generate_manifest(test_resources)
+  assertion_code = generate_assertions(test_resources)
+
+  create_sitepp(master, tmp_environment, sitepp_content)
+
+  step "run agent in #{tmp_environment}, run all assertions" do
+    with_puppet_running_on(master,{}) do
+      agents.each do |agent|
+        on(agent, puppet("agent -t --debug --trace --show_diff --server #{master.hostname} --environment #{tmp_environment}"),
+           :accept_all_exit_codes => true) do |result|
+          assert(result.exit_code==2,'puppet agent run failed')
+          run_assertions(assertion_code, result)
+        end
+
+        if agent['platform'] =~ /fedora|centos|el|redhat|scientific|ubuntu|debian|cumulus/
+          step "assert no redacted data in syslog" do
+            #sigh.  gee, thanks for the help, beaker!
+            result = dump_puppet_log(agent)
```

Comment:
the above changed that.  not the best fix, and i suppose we can write our own helper for grabbing the puppet log on various platforms.  but for now this works for now, based on copy, paste of the platforms supported in beaker.

i'd prefer to leave it like this.  but i can re-write if people think it's important in this sprint